lib/stronger_parameters/controller_support/permitted_parameters.rb

```diff
@@ -0,0 +1,169 @@
+require 'stronger_parameters/constraints'
+
+module StrongerParameters
+  module ControllerSupport
+    module PermittedParameters
+      def self.included(klass)
+        klass.extend ClassMethods
+        klass.before_action :permit_parameters
+      end
+
+      class PermittedParametersHash < HashWithIndifferentAccess
+        def initialize(other = nil)
+          super()
+          merge!(other) unless other.nil?
+        end
+
+        def merge!(other)
+          return self if other.nil?
```

Comment:
that would conflict with Hash#deep_merge! (used below, line 88), unless we renamed to something specific?